src/core/polyhedral/codegen_llvm.cc

```diff
@@ -338,9 +351,48 @@ class CodeGen_TC : public Halide::Internal::CodeGen_X86 {
       functionPassManager.doFinalization();
       modulePassManager.run(*module);
 
-      if (FLAGS_llvm_dump_after_opt) {
-        module->print(llvm::dbgs(), nullptr, false, true);
+      LOG_IF(INFO, FLAGS_llvm_dump_after_opt)
+          << "[LLVM-IR] After optimization:\n"
+          << toString(module.get());
+    }
+
+    // In a separate process call the LLC and dump the ASM
+    // Since this is for debugging purposes only atm, separate process is fine.
+    if (FLAGS_llvm_dump_asm) {
+      // Create a temporary file (do not use racy tmpname;
+      // do not use tmpnam that immediately destroys the file).
+      char fileCName[256] = "/tmp/XXXXXXXXXX";
+      auto fd = mkstemp(fileCName);
+      ScopeGuard sg([fileCName]() { unlink(fileCName); });
+      CHECK_GE(fd, 0) << "mkstemp(): " << fileCName << " returned fd: " << fd;
+      close(fd);
+
+      // Close and reopen as FILE*
+      auto tmp = fopen(fileCName, "wb+");
+      auto llvmIr = toString(module.get());
+      std::fputs(llvmIr.c_str(), tmp);
+      fclose(tmp); // Close properly so the piped command can open without race
+
+      std::string fileName(fileCName);
+      // TODO: evolve this in a better way to find LLVM
+      auto command = std::string(" `which llc || which llc-5.0` ") +
```

Comment:
can this be rather handled at cmake level? doing this here increases chances of spurious build failures that people might encounter

Reply:
Yes, that's why it's marked TODO. 

One possible way is to have a CLANG_PREFIX macro in the same way we have CUDA_HOME / CUB_HOME.
However I am not sure yet what the best solution is because we need to experiment a bit in our build system where setting CUDA_HOME / CUB_HOME seems to be problematic.

In any case, realize that this is completely unrelated to build failures. 
This **may** result in runtime errors when passing ```--llvm_dump_asm=1``` if the user does not have clang in his/her path.